.github/workflows/unitctl.yml

```diff
@@ -0,0 +1,132 @@
+
+name: unitctl
+
+on:
+  pull_request:
+    paths:
+      - tools/unitctl/**
+  push:
+    branches:
+      - master
+    tags:
+      - unitctl/[0-9]+.[0-9]+.[0-9]+
+
+permissions:
+  contents: write
```

Comment:
What exactly is given write permissions to?

Reply:
Allows the job to create or update the release. It's opt-in now.

> Work with the contents of the repository. For example, contents: read permits an action to list the commits, and contents: write allows the action to create a release. For more information, see "[Permissions required for GitHub Apps](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents)."

Reply:
So it doesn't have write access to the repository itself?

Reply:
It allows the action to write to any of these endpoints: https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents

Reply:
Wow!

Is there anyway to give more fine-grained access?